app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PostsController < ApplicationController
+
+  def index
+    # friends = current_user.friends.pluck(:friend_id)
+    # friends << current_user.id
+    @comment = Comment.new
+    @posts = Post.all.user_friends(current_user).with_comments_and_likes
```

Comment:
What do you think about transferring complex logic such as this in the model? We can create an instance method in either the `user` or a class method/scope in the `post` model to handle this. As much as possible we wanna follow the principle of **thin-controller fat model**. For instance if we transfer this advanced querying in the `user` model we can simply do this:

`@posts = current_user.advanced_query_method_that_was_transferred_to_the_user_model`